Added SnakeGame

todorzag opened the pull request.

todorzag committed:
Added SnakeGame

todorzag committed:
Added AppleSpawner functionality

todorzag committed:
Changed while loop to do while loop

todorzag committed:
Added missing files, Game Class, GameStart logo

todorzag committed:
Added aesthetic Logos and Score

todorzag committed:
Rearranged GetInstance method on Game.cs

todorzag committed:
Remade SnakeGame to have the console window as a border

todorzag commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,143 @@
+            Console.Clear();
+            Console.WriteLine(Logos.GameStartLogo);
+
+            Thread.Sleep(2000);
```
How to move Thread.Sleep() to main?

todorzag committed:
Added UML

todorzag committed:
Made new logic for SnakeGame to remove GameBoard

todorzag committed:
Moved applespawner method to Game and Game Start to Main

todorzag committed:
Removed apple spawner instance from Game

todorzag committed:
Remade UML to be in line with new changes

todorzag committed:
Moved CheckIfOutOfBounds And Teleport from SnakePart to Snake, Add ConsoleHeight and ConsoleWidth as parameters, Added some unit tests

todorzag committed:
Removed unnecessary usings and remade UML to reflect new changes

todorzag committed:
Added Teleport method unit tests and property for snake current position

todorzag committed:
Moved Turn methods from SnakePart to Snake and changed unit tests to reflect changes

todorzag committed:
Remade UML to reflect new changes

todorzag committed:
Changed Made:
 - Moved Turn logic From Snake to Game
 - Added enum Directions and Dict movementDirections
 - New method to check if ShouldTeleport
 - Added logic for game over in Program
 - Made SnakeHead public
 - Moved onApple to SnakePart
 - Removed changePosition from SnakePart

todorzag commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,221 @@
+            Down
+        }
+
+        public delegate void Del();
```
Need to remove

todorzag committed:
Removed Delagate Del, Added check for keypressed direction

todorzag committed:
Added _lastDirection and remade DirectionHandler to work properly

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,156 @@
+        {
+            SnakeHead = new SnakePart(1, 4, "○");
+
+            _snakeBody = new List<SnakePart>
```
What if we want to make starting length of the snake configurable

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,156 @@
+
+        public Snake()
+        {
+            SnakeHead = new SnakePart(1, 4, "○");
```
Isn't the head part of the body

mvlt commented on SnakeGame/SnakeGame/SnakePart.cs:
```diff
@@ -0,0 +1,27 @@
+        public string Symbol { get; }
+        public (int, int) Position { get => (X, Y); }
+
+        public SnakePart(int x, int y)
```
Do we need 2 constructors for different type of displaying. Snake part should be able to print itself differently based on a param

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,156 @@
+
+        public void MoveX(int x)
+        {
+            SnakeHead.X += x;
```
Why don't we check here if the move is appropriate?

mvlt replied:
Also do the teleport here

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,156 @@
+            SnakeHead.Y += y;
+        }
+
+        public void UpdateBodyPosition()
```
Another way which we could try is adding a new element in front and removing the last one.

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,234 @@
+            };
+        }
+
+        private bool ShouldEndGame()
```
Shouldn't this be part of the Snake class

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,234 @@
+            _snake.ClearLastSnakePart();
+            _snake.UpdateBodyPosition();
+
+            moveMethod(directionNum);
```
Better use try catch and move the logic for teleport and game over checks in snake class

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,156 @@
+{
+    public class Snake
+    {
+        public SnakePart SnakeHead;
```
In Snake class shouldn't be just named Head (Snake.Head not Snake.SnakeHead)

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,234 @@
+            RenderScore();
+        }
+
+        private void PopulateDirections()
```
Why do we initialize it like this

todorzag committed:
Changes Made:
 - movementDirections is now initialized in it's field
 - Renamed SnakeIsAlive to HasEnded
 - Moved ShouldEndGame And ShouldTeleport into snake
 - Made a method to GenerateSnakeBody by length
 - Added try catch logic to MoveSnake
 - Removed one constructor from SnakePart
 - Renamed SnakeHead to Head

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,231 @@
+            return false;
+        }
+
+        public bool OnApple((int,int) applePosition)
```
Do you need this method

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,231 @@
+            return snakeParts;
+        }
+
+        public void RenderSnake()
```
Can we handle the logic for different symbols of the snake in render method

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,231 @@
+            _snakeBody[0].Y = Head.Y;
+        }
+
+        public bool ShouldEndGame()
```
Need better naming

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,231 @@
+        {
+            ClearLastSnakePart();
+
+            for (int i = _snakeBody.Count - 1; i > 0; i--)
```
Is there better way for that job

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,231 @@
+        {
+            bool result = false;
+
+            if (CheckIfOutOfBounds())
```
Similar logic like shouldTeleport

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,184 @@
+        {
+            Directions direction;
+            
+            if (IsKeyInDictionary(keyPressed))
```
Why wrapped in function

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,184 @@
+            }
+            else
+            {
+                direction = _lastDirection;
```
Seems a little hacky try other way

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,184 @@
+
+                _applePosition = (x, y);
+
+                if (!_snake.CheckSpawnOnSnake(_applePosition))
```
Why wrapped

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,184 @@
+
+                DirectionHandler(_keyPressed);
+            }
+            while (!HasEnded);
```
Here also checking hasEnded, so why not end here but call Over in program.cs

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,231 @@
+            return applePosition == Head.Position;
+        }
+
+        public void EatApple()
```
Change the name to tell what the function is doing

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,184 @@
+        private string _user;
+        private (int, int) _applePosition;
+        private Directions _lastDirection;
+        private string _keyPressed = "RightArrow";
```
Why do you create enum and dictionary and hardcoded here

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,184 @@
+
+        public bool HasEnded = false;
+
+        private Dictionary<string, Directions> movementDirections 
```
Do we need a dictionary? Could we define the enum other way so we don't need this dictionary?

mvlt reviewed (changes requested):
Think also about interfaces for some of the classes

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,231 @@
+            }
+        }
+
+        public void ClearLastSnakePart()
```
I was expecting to remove it from snake body

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,231 @@
+
+        public void Teleport()
+        {
+            int x = Head.X;
```
Head.X is more explainable than x why assigning to a new variable

todorzag committed:
Changes Made:
 - Removed enum Directions and added _legalKeys in its place
 - Added LegalKeyHandler
 - Added WaitForKeyPress
 - Create Interface for SnakePart
 - Remade logic for RenderSnake to render different symbols
 - Renamed some variables and methods for better readability

todorzag committed:
Changes Made:
 - Made new Thread in Program for Game
 - Added SetLegalKeyPressed Method
 - Remade MoveSnake logic with while loop
 - Renamed MoveEndsGame to IsNotPossibleMove

todorzag committed:
Small error

todorzag committed:
Remade _legalKeys and it's logic to enum Directions

todorzag committed:
Renamed variables

todorzag committed:
Fixed small bug where GameOver didn't execute when pressing ESC to exit

todorzag committed:
Added Directions enum, removed keylogger Task and refactored MoveSnake to work without Task

todorzag committed:
Removed default from DirectionHandler

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,203 @@
+    {
+        public bool SnakeIsAlive = true;
+
+        private Snake _snake = new Snake(4);
```
Better user to be able to choose starting length ex. from 0 to 10

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,202 @@
+        private (int, int) _lastBodyPartPosition;
+        private int _startingLenght;
+
+        private int _consoleHeight = Console.WindowHeight;
```
why do we need to store height and width in fields

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,202 @@
+            return snakeParts;
+        }
+
+        private bool MoveIsNotPossible(bool hasWalls)
```
Not the best name for this function

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,203 @@
+
+            GameLoop();
+
+            if (!SnakeIsAlive)
```
Do we need to check if Snake is Alive here

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,203 @@
+            }
+        }
+
+        private void GameOver()
```
Game over not best name if also is saving the score

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,203 @@
+            RenderScore();
+        }
+
+        private void AppleHandler()
```
AppleHandler is also adding snake parts

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,203 @@
+
+            OpenStartingMenu();
+            // Close starting menu
+            Console.Clear();
```
Can we clear the console in some of the functions which are called in the Main Process

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,203 @@
+            }
+        }
+
+        private void StartingProcesses()
```
Why do we need to spawn apple and render the snake here

mvlt commented on SnakeGame/SnakeGame/Game.cs:
```diff
@@ -0,0 +1,203 @@
+            _hasWalls = AskHasWalls();
+        }
+
+        private void WaitForKeyPress()
```
Waiting for key press is also writing starting logo

mvlt commented on SnakeGame/SnakeGame/Snake.cs:
```diff
@@ -0,0 +1,202 @@
+            return result;
+        }
+
+        private bool CheckTeleport(bool hasWalls)
```
better name

todorzag committed:
Changes made:
 - Removed StartingProcess
 - Moved Console.Clear() into OpenStartingMenu
 - Added functionality where user enters the starting length of the snake
 - Moved SaveHighScore out of GameOverScreen
 - Remade logic for AppleHandler to only return bool
 - Renamed AppleHandler to IsOnApple
 - Moved logic for AppleHandler in MoveSnake
 - Removed fields for Console height and width
 - Renamed MoveIsNotPossible to MoveIsIllegal

todorzag committed:
Added ScoreManager and DiffilcultyHandler, Remade Game logic to work with new classes

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,219 @@
+        public bool SnakeIsAlive = true;
+
+        private Snake _snake;
+        private IDiffilcultyHandler _diffilcultyHandler = new DiffilcultyHandler();
```
Syntax error. Violation of Dependency Inversion Principle.

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,219 @@
+        private IDiffilcultyHandler _diffilcultyHandler = new DiffilcultyHandler();
+        private IScoreManager _scoreManager = new ScoreManager();
+
+        private string _user;
```
Violation of Single Responsibility Principle.

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,219 @@
+            }
+        }
+
+        private void OpenStartingMenu()
```
SRP

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,219 @@
+
+            if (_snakeLength < 0 || _snakeLength > 10)
+            {
+                throw new ArgumentOutOfRangeException
```
SRP

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,219 @@
+            Console.WriteLine(Logos.GameStartLogo);
+        }
+
+        private void SpawnApple()
```
SRP

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,219 @@
+
+        private bool IsOnApple()
+        {
+            return _applePosition == _snake.Head.Position;
```
Is this responsibility for the game or the snake

mvlt commented on SnakeGame/SnakeGame/Classes/Snake.cs:
```diff
@@ -0,0 +1,191 @@
+            return snakeParts;
+        }
+
+        private bool MoveIsIllegal(bool hasWalls)
```
This name sounds more like a statement than a function

mvlt commented on SnakeGame/SnakeGame/Classes/Snake.cs:
```diff
@@ -0,0 +1,191 @@
+            return _body.Any(x => x.Position == applePosition);
+        }
+
+        public void MoveX(int x, bool gameHasWalls)
```
x ?

todorzag committed:
WIP ChangesMade:
 - Added GameConfigCapturer
 - Remade Game constructor to work with GameConfigCapturer
 - Added Factory to initialize class interfaces
 - Added Menu
 - Added Validator
 - Added Writer
 - Added MoveChecker
 - Remade MoveX and MoveY to work with MoveChecker
 - Remade Teleport to work with MoveChecker
 - Added FoodSpawner
 - Added Player
 - Added constructor for ScoreManager and DiffilcultyHandler
 - Remade SnakePart to Point
 - Created Interfaces for Classes

todorzag committed:
WIP Changes Made:
 - Added ISpawnable interface
 - Added Dollar and Apple spawnables
 - Reworked FoodSpawner to work with spawnables logic
 - Reworked Game to work with spawnables logic
 - Added Timer to Game for Dollar spawn
 - Added try catch logic for snake length exception
 - Added Apple and Dollar to Factory

todorzag committed:
Added default value to DiffilcultyHandler and ScoreManager

todorzag committed:
Changes Made:
 - Reworked Spawner and Spawnables
 - Added GameConfig and IGameConfig
 - Reworked GameConfigCapturer to return IGameConfig
 - Reworked Game to work with new logic

todorzag committed:
Fixed small bug

todorzag committed:
Changes Made
 - Renamed ISpawnable to IBonus
 - Renamed ISnakePart to IPoint
 - Added ScoreValue to IBonus
 - Added private field for IGameConfig in Game
 - Reworked Game to work with IGameConfig
 - Moved DirectionHandler to Snake
 - Made custom exception for game end
 - Added ScoreRange for Score Render in Spawner

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,149 @@
+    {
+        public bool snakeIsAlive = true;
+
+        private IScoreManager _scoreManager =
```
We could pass scoreManager as a parameter to the Game class and initialize it in the constructor

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,149 @@
+        private IScoreManager _scoreManager =
+            Factory.CreateScoreManager();
+
+        private IDiffilcultyHandler _diffilcultyHandler =
```
DifficultyHandler also

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,149 @@
+
+            Menu.StartingScreen();
+
+            _snake = Factory.CreateSnake(_config.SnakeLength);
```
Snake could also be passed to the constructor

mvlt commented on SnakeGame/SnakeGame/Classes/Snake.cs:
```diff
@@ -0,0 +1,156 @@
+        public IPoint Head { get => Body[0]; }
+        public List<IPoint> Body { get; private set; }
+
+        public (int, int) CurrentPosition
```
Are we using this only in the tests

mvlt commented on SnakeGame/SnakeGame/Classes/SnakeMoveChecker.cs:
```diff
@@ -0,0 +1,54 @@
+            bool isValid = true;
+
+            // добави exception class
+            if (CheckIfOutOfBounds(newPosition) && !gameHasWalls)
```
If game has no walls no need to evaluate CheckIfOutOfBounds

mvlt commented on SnakeGame/SnakeGame/Utils/ScoreManager.cs:
```diff
@@ -0,0 +1,36 @@
+            Writer.WriteAt(0, 0, $"Score: {Score}");
+        }
+
+        public void RenderLogo()
```
We could have something like fancyMessageRenderer which takes the score and do the rendering of all logos. I think score manager should not be responsible for that also

mvlt commented on SnakeGame/SnakeGame/Utils/StandardMessages.cs:
```diff
@@ -0,0 +1,37 @@
+
+namespace SnakeGame.Utils
+{
+    public class StandardMessages
```
We are not only storing the message content but we also writing them better name for the class

todorzag committed:
WIP Bonuses

todorzag committed:
Changes made:
 - Remade Game constructor
 - Added strategy design pattern to bonuses
 - Added BonusesHandler

todorzag committed:
Changes made:
 - Renamed IOnConsume to IConsumable
 - Added Strategies for IConsumable
 - Fixed Spawner
 - Moved Utility initialization in Factory.CreateGame

todorzag committed:
Changes Made:
 - Added SnakeGame.csproj
 - Reworked GameConfig to be static
 - Added end game for ScoreUnderZero
 - Changed Cross ScoreValue from -5 to -30
 - Added Methods for timers in Game
 - Added private list for timers in Game
 - Moved ScoreManager.Render in if statement in OnAction to fix bug where score randomly appeared in the middle of the screen
 - Made all 'new' instances to Factory methods in Snake

todorzag committed:
Now Added SnakeGame.csproj

todorzag committed:
Made all strategies in one class

todorzag committed:
Reworked arguments for TimerCallback

todorzag committed:
Small Changes

todorzag committed:
Changes Made:
 - Returned strategies to separate classes
 - Reworked Writer to WriteMessage and Files
 - Removed Separate classes for Bonuses
 - Made base class Bonus for all Bonuses
 - Reworked ScoreManager to set Score from Snake Length
 - Fixed bug in Spawner where if wasn't working correctly
 - Reworked StandardMessages to only return messages
 - Reworked BonusesHandler to be inline with ScoreManager changes
 - Made timers from game into separate tasks
 - Added new SwitchDirectionStrategy And Bonus
 - Moved Direction from Game to Snake

todorzag committed:
Small changes

todorzag committed:
Fixed Score negative number exception

todorzag committed:
Fixed Render bug

todorzag committed:
Renamed Bonuses To Foods

todorzag committed:
Changed made:
 - Renamed all Bonus files to Food
 - Added OnRespawn Strategies to Food
 - Added TimeDelay to Food for timed spawns
 - Added isRendering bool in Writer to fix bug

todorzag committed:
Version with timers and no OnRespawn Strategy

todorzag committed:
Changes made:
 - Fixed bug in DiffilcultyHandler where it didn't work
 - Added CheckToDecreaseLevel in case score returns under set amount
 - Added ScoreValue to Food
 - Added GetSafeDirection in SnakeMoveChecker for direction switching
 - Fixed bug where snake parts were randomly rendered (in Writer)
 - Added PreviousScore in ScoreManager to ease DiffilcultyHandler checks

todorzag committed:
Changes Made:
 - Reworked FindSafeDirection
 - Removed GameEndException and remade logic for where it was

mvlt commented on SnakeGame/SnakeGame/Classes/Food.cs:
```diff
@@ -0,0 +1,42 @@
+
+namespace SnakeGame.Classes
+{
+    public class Food : Point, IFood
```
Renderable?

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,136 @@
+        {
+            Console.CursorVisible = false;
+
+            ScreenRenderer.StartingScreen();
```
StartingScreen no verb in the name of the function

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,136 @@
+            GameLoop();
+
+            Writer.FileWrite(GameConfig.Player, _scoreManager.CurrentScore);
+            ScreenRenderer.GameOverScreen(_scoreManager);
```
No verb in GameOverScreen and why need the dependence to scoreManager

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,136 @@
+        private void EnableTimers()
+        {
+            _timers.Add(new Timer((e)
+                => TimerCallback(Factory.CreateSwitch), null, 10000, 10000));
```
Magic numbers

mvlt commented on SnakeGame/SnakeGame/Utils/ScoreManager.cs:
```diff
@@ -0,0 +1,42 @@
+            Writer.ConsoleWriteAt(0, 0, $"Score: {CurrentScore} ");
+        }
+
+        public void CheckScoreUnderZero(ISnake snake)
```
Do we need scoreManager to be dependent of the snake?

mvlt commented on SnakeGame/SnakeGame/Utils/ScreenRenderer.cs:
```diff
@@ -0,0 +1,41 @@
+
+        public static void GameOverScreen(IScoreManager scoreManager)
+        {
+            Console.WriteLine(Logos.GameOverLogo);
```
Why not using the writer?

mvlt commented on SnakeGame/SnakeGame/Utils/Validator.cs:
```diff
@@ -0,0 +1,22 @@
+
+namespace SnakeGame.Utils
+{
+    public class Validator 
```
This seems more like snakeLengthValidator

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,136 @@
+                        _snake,
+                        _scoreManager);
+
+                    _scoreManager.Set(_snake.Body, _diffilcultyHandler);
```
Why scoreManager also need to know about defficultyHandler

mvlt commented on SnakeGame/SnakeGame/Classes/Snake.cs:
```diff
@@ -0,0 +1,182 @@
+            Body.Add(Factory.CreatePoint(x, y));
+        }
+
+        public void RemovePart()
```
Remove tail better. With name RemovePrat I have the feeling that with this function I can remove every part but I can't

mvlt commented on SnakeGame/SnakeGame/Classes/Snake.cs:
```diff
@@ -0,0 +1,182 @@
+
+        public void RemovePart()
+        {
+            ClearLastBodyPart();
```
RemovePart is calling ClearLastBodyPart with this naming I am assuming they are doing same thing

mvlt commented on SnakeGame/SnakeGame/Classes/Snake.cs:
```diff
@@ -0,0 +1,182 @@
+            switch (Direction)
+            {
+                case Directions.RightArrow:
+                    MoveY(1);
```
Magic number

mvlt commented on SnakeGame/SnakeGame/Classes/Snake.cs:
```diff
@@ -0,0 +1,182 @@
+
+            var isValid = SnakeMoveChecker.Check(this, _newPosition);
+
+            if (isValid == true)
```
Boolean expression with three cases?

mvlt commented on SnakeGame/SnakeGame/Utils/SnakeMoveChecker.cs:
```diff
@@ -0,0 +1,82 @@
+    internal class SnakeMoveChecker
+    {
+        public static bool? Check(
+            ISnake snake, IPoint newPosition)
```
Do we need the whole snake object

mvlt commented on SnakeGame/SnakeGame/Utils/Writer.cs:
```diff
@@ -0,0 +1,44 @@
+            while (true)
+            {
+                if (_isRendering == false)
+                {
```
This logic seems strange. Is there more simple approach?

mvlt commented on SnakeGame/SnakeGame/Interfaces/IOnConsume.cs:
```diff
@@ -0,0 +1,14 @@
+
+namespace SnakeGame.Interfaces
+{
+    public interface IOnConsume
```
Consumable?

mvlt reviewed (commented):
Do we need all implemented interfaces

todorzag committed:
Fixed some of previous issues

todorzag committed:
Changes made:
 - Introduced local variables for milliseconds in Timers
 - Removed FoodHandler dependency for ScoreManager
 - Removed MaxLength property from Snake
 - Introduced local variables for Forward and Backward in Snake
 - Introduced enum for States in SnakeMoveChecker
 - Reworked SnakeMoveChecker.Check to only need Snake.Body
 - Reworked SnakeLengthValidator.ValidateMaxSnakeLength to only need Snake.Body

todorzag committed:
Renamed Validator methods

todorzag committed:
Renamed ScreenRenderer methods

todorzag committed:
Changes made:
 - Reworked DiffilcultyHandler.CheckToChangeLevel method to not need PreviousScore
 - Moved ValidateMaxLength check into Game
 - Introduced consts for LevelBreakPoint and LevelSpeedModifier

todorzag committed:
SQL commit

todorzag committed:
Added previously unsolved tasks

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,136 @@
+        {
+            Console.CursorVisible = false;
+
+            ScreenRenderer.StartingScreen();
```
Again no verb in the name of the function

mvlt commented on SnakeGame/SnakeGame/Classes/Game.cs:
```diff
@@ -0,0 +1,136 @@
+            GameLoop();
+
+            Writer.FileWrite(GameConfig.Player, _scoreManager.CurrentScore);
+            ScreenRenderer.GameOverScreen(_scoreManager);
```
And here why marked as resolved

todorzag replied:
Sorry my bad, confusion on my part because of Menu.cs

mvlt commented on SnakeGame/SnakeGame/Classes/Strategies/AddSnakePartStrategy.cs:
```diff
@@ -0,0 +1,21 @@
+
+namespace SnakeGame.Classes.Strategies
+{
+    public class AddSnakePartStrategy : IConsumable
```
How many times this strategy will be instantiated?

mvlt commented on SnakeGame/SnakeGame/Constants/Logos.cs:
```diff
@@ -0,0 +1,122 @@
+            "\r\n█▄▄▄█ ▀▀▀ ▀▀▀▀ ▀ ▀▀ ▀▀▀ ▀";
+        }
+
+        public static string GenerateScoreLogo(int scoreInt)
```
Should this functionality be part of Logos class

todorzag committed:
Changes made:
 - Removed Menu class
 - Reworked all strategies to singleton pattern
 - Moved GenerateScoreLogo to new class
 - Returned if else in Writer.ConsoleWriteAt to while because of StackOverflow Exception

mvlt committed:
Merge pull request #22 from todorzag/SQL

SQL commit

todorzag committed:
Added DiscoMode

todorzag committed:
Merge branch 'SnakeGame' of https://github.com/todorzag/c-sharp-oop into SnakeGame
